prometheus-net/Histogram.cs

```diff
@@ -106,28 +85,15 @@ public void Observe(double val)
                     return;
                 }
 
-                int bucketIndex = -1;
-                for (int i = 0; i < _buckets.Length; i++)
+                for (int i = 0; i < _upperBounds.Length; i++)
                 {
-                    if (val <= _buckets[i])
+                    if (val <= _upperBounds[i])
                     {
-                        bucketIndex = i;
+                        _bucketCounts[i].Add(1);
                         break;
                     }
                 }
-
-                if (bucketIndex == -1)
-                {
-                    Trace.WriteLine(string.Format("Couldn't find a bucket for {0} in metric {1}", val, ToString()));
-                    return;
-                }
-
-                lock (_lock)
-                {
-                    ++_count;
```

Comment:
See previous explanation about why count shouldn't be tracked here.
It can be computed as we traverse to the bucket counts to produce a more internally consistent representation of the count as we are producing the result of a particular collection/populate action.